.github/workflows/build.yml

```diff
@@ -104,8 +104,33 @@ jobs:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
     env:
-      # Set up wheels matrix.  This is CPython 3.6--3.9 for all OS targets.
-      CIBW_BUILD: "cp3[6-9]-*"
+      # Set up wheels matrix.  This is CPython 3.6--3.10 for all OS targets.
+      CIBW_BUILD: "cp36-* cp37-* cp38-* cp39-* cp310-*"
+      # We build with oldest-supported-numpy, see:
+      # https://github.com/scipy/oldest-supported-numpy/blob/master/setup.cfg
+      CIBW_BEFORE_BUILD: "pip install oldest-supported-numpy"
```

Comment:
I think `oldest-supported-numpy` isn't right - I think we _require_ Numpy 1.16+, but Numpy 1.13 had support for 1.16.

You could potentially add a "build-constraints.txt" file, and add that as a `-c` option for CI wheel builds?

Reply:
Maybe I can publish another PyPI package called `oldest-supported-numpy-greater-than-or-equal-to-116`. :/

Reply:
I'm not actually sure that recent versions of pip will honour the installed numpy when building wheels. :/

Reply:
They shouldn't respect it - that's kind of the point of build isolation!  But there should be a CIBW flag to let you pass through a constraints file to the build command.

Reply:
There is, but it doesn't work. It's documented as not having an effect on linux builds -- https://cibuildwheel.readthedocs.io/en/stable/options/#dependency-versions -- although it used on linux builds, but only for the test dependencies -- https://github.com/pypa/cibuildwheel/blob/main/cibuildwheel/linux.py#L251.

I suspect the only option is to hardcode the numpy version into `build-system.requires` and force everyone to use it and to not use pip if they want a different version. I don't know if that will break our conda-forge feedstock though.

Reply:
It seems like it may have _some_ effect:

https://github.com/pypa/cibuildwheel/blob/52f7801799043bd5a734617925c5b9542ef9c81d/cibuildwheel/linux.py#L141-L148

I'd have thought that things like `pip` and `setuptools` are pinned by `manylinux2010`, but Numpy shouldn't be (else how would Numpy built their wheels?).

Reply:
(Oh yeah, I see now that this variable doesn't actually appear in the build command.)

Reply:
I created a "solution" by copying the pyproject.toml file. :/